appsmithorg/appsmith
# fix: trigger fields linting issue

Fixes #11943

## Description
- DataTreeEvaluator - Include trigger fields(onClick etc...) in dependency map to keep them in sync with linting. #10941

- Added `loadingProperties` in table widget config (can be configured for other widgets in future)
  - this is a list of regex that is used to determine, which properties of a widget can trigger it's loading state
  - when undefined, any path bound to a API/Query can trigger the loadingState (same as existing behaviour)

- [performance logs](https://github.com/appsmithorg/appsmith/pull/11875/commits/63acf2ba421d18069cf00e409b17f6ac1972bf7d) checked with A-force app (current branch vs release)
  - increase in time per update, ranged between 30ms - 70ms 

## Notes
Brings back changes from #10941
Which was reverted (here #11163) to address #11136


## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Create a button and an API.
- Bind `{{Api1.run()}}` to onClick event of the button.
- Refresh the page and notice `Api1 is undefined` error doesn't show up anymore.
- Delete `Api1` and notice the error is shown on onClick field.
- Create `Api1` again and notice the error goes away.

## Checklist:











## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>


    // Code coverage diff between base branch:release and head branch: fix/onclick-linting-issue 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 55.67 **(0)** | 37.07 **(0.02)** | 35.26 **(0)** | 55.93 **(0)**
 :red_circle: | app/client/src/utils/WidgetFactory.tsx | 74.49 **(0.02)** | 73.68 **(1.46)** | 61.11 **(-3.6)** | 74.49 **(0.02)**
 :green_circle: | app/client/src/utils/WidgetLoadingStateUtils.ts | 96.83 **(0.16)** | 92 **(2)** | 100 **(0)** | 98.08 **(0.12)**
 :red_circle: | app/client/src/widgets/BaseWidget.tsx | 79.07 **(-0.46)** | 44.44 **(0)** | 73.53 **(-2.23)** | 78.23 **(-0.46)**
 :green_circle: | app/client/src/widgets/TableWidget/widget/index.tsx | 30.07 **(0.33)** | 9.43 **(0)** | 30.19 **(1.34)** | 29.77 **(0.37)**
 :red_circle: | app/client/src/workers/DataTreeEvaluator.ts | 75.93 **(-0.4)** | 57.47 **(-0.2)** | 73.86 **(-0.85)** | 75.5 **(-0.55)**</details>
